src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

```diff
@@ -125,7 +130,7 @@ public override async ValueTask DisposeAsync()
             }
             catch (Exception ex)
             {
-                _trace.LogError(0, ex, $"Unexpected exception in {nameof(SocketConnection)}.{nameof(Start)}.");
+                _trace.LogError(0, ex, $"Unexpected exception in {nameof(SocketConnection)}.");
             }
             finally
             {
```

Comment:
I should have caught this on #30771, but we should never be disposing `_sender` in this class now that it's coming from a pool.

Reply:
Nvm. I see the logic now. It's only disposing `_sender` if `SendAsync` throws in theory.